actgen/plot.py

```diff
@@ -1,178 +1,131 @@
-	if len(g_difference.shape) == 1:
-		return time_step, g_difference, 0  # if only 1 example
-	avg_g_difference = np.mean(g_difference, axis=0)
-	ci = 1.96 * np.std(g_difference, axis=0) / math.sqrt(len(g_difference))
-	return time_step, avg_g_difference, ci
-
-
-def preprocess_training_rewards(directory):
+			all_file_paths.append(file_path)
+	return all_file_paths
+
+
+def preprocess_data(experiment_name, file_ending):
 	"""
-	go through the specified directory and compute the data needed for plotting
-	return time_step, rewards, confidence_interval
+	get all the csv data from a single gym environment experiment from all types of agents
+	return a pandas Dataframe object
 	"""
-	if not os.path.exists(directory):
-		logging.info("directory {} not found".format(directory))
-		return [], [], []
-	time_step = np.array([])
-	rewards = np.array([])
-		# iterate over the directory
-	for file_name in os.listdir(directory):
-		# find all the saved gscore files
-		if file_name.endswith("training_reward.csv"):
-			file_path = os.path.join(directory, file_name)
-			step, r = read_csv(file_path)
-			if len(time_step) == 0:
-				time_step = step
-			assert time_step.all() == step.all()
-			rewards = r if len(rewards) == 0 else np.vstack([rewards, r])
-	# average the g_difference and get 95% confidence interval
-	# each row of g_difference contains an example
-	if len(rewards) == 0:
-		raise RuntimeWarning("no training reward csv files found in the specified tag directory")
-	if len(rewards.shape) == 1:
-		return time_step, rewards, 0  # if only 1 example
-	avg_rewards = np.mean(rewards, axis=0)
-	ci = 1.96 * np.std(rewards, axis=0) / math.sqrt(len(rewards))
-	return time_step, avg_rewards, ci
-
-
-def preprocess_training_loss(directory):
+		# gather all files from agent
+	n_dup_agent = experiment_name
+	n_dup_files = iterate_through_directory(n_dup_agent, file_ending)
+
+	n_dup_oracle_agent = experiment_name + "-oracle"
+	n_dup_oracle_files = iterate_through_directory(n_dup_oracle_agent, file_ending)
+
+	no_dup_agent = experiment_name + "-nodup"
+	no_dup_files = iterate_through_directory(no_dup_agent, file_ending)
+
+	rand_actions_agent = experiment_name + "-rand"
+	rand_actions_files = iterate_through_directory(rand_actions_agent, file_ending)
+
+	rand_actions_oracle_agent = experiment_name + "-rand-oracle"
+	rand_actions_oracle_files = iterate_through_directory(rand_actions_oracle_agent, file_ending)
+
+	# gather all csv data from the files
+	fnames = n_dup_files + n_dup_oracle_files + no_dup_files + rand_actions_files + rand_actions_oracle_files
+	agents = ["N dup"] * len(n_dup_files) + \
+				["N dup with oracle"] * len(n_dup_oracle_files) + \
+				["no dup"] * len(no_dup_files) + \
+				["random actions"] * len(rand_actions_files) + \
+				["random actions with oracle"] * len(rand_actions_oracle_files)
+	data = pd.concat([read_csv(fname, agent) for fname, agent in zip(fnames, agents)])
```

Comment:
I think it's probably easier to get the `pd.DataFrame` for each set of data separately (above), and for each one, you can simply do `df['agent'] = "N dup with oracle"` or the equivalent. This should set the agent field for all the rows, and you don't have to count anything. Then you can `pd.concat` the resulting `DataFrames`.